Lingua-App/Lingua/Lingua/Model/Project.swift

```diff
@@ -17,21 +17,42 @@ struct Project: Identifiable, Hashable, Equatable, Codable {
   var title: String
   var swiftCode: SwiftCode
   var swiftCodeEnabled: Bool = true
+  var createdAt: Date
+  var lastLocalizedAt: Date?
   
   init(id: UUID,
        type: LocalizationPlatform,
        apiKey: String = "",
        sheetId: String = "",
        directoryPath: String = "",
        title: String = "",
-       swiftCode: SwiftCode = .init(stringsDirectory: "", outputSwiftCodeFileDirectory: "")) {
+       swiftCode: SwiftCode = .init(stringsDirectory: "", outputSwiftCodeFileDirectory: ""),
+       createdAt: Date = .init(),
+       lastLocalizedAt: Date? = nil) {
     self.id = id
     self.type = type
     self.apiKey = apiKey
     self.sheetId = sheetId
     self.directoryPath = directoryPath
     self.title = title
     self.swiftCode = swiftCode
+    self.createdAt = createdAt
+    self.lastLocalizedAt = lastLocalizedAt
+  }
+  
+  init(from decoder: Decoder) throws {
```

Comment:
Let's add a comment on why we are using a custom decoder here - for the sake of persistence through the installs and control.